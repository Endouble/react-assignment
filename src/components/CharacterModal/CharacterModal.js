import React from 'react';
import PropTypes from 'prop-types';

import { Modal } from 'semantic-ui-react';

const CharacterModal = (props) => {
    const { onCloseCallback, character } = props;
    return (
        <Modal open closeIcon onClose={onCloseCallback}>
            <Modal.Header>
                {character.name}
            </Modal.Header>
            <Modal.Content>
                <Modal.Description>
                    <p id="gender">
                        Gender:
                        {character.gender}
                    </p>
                    <p id="eyes">
                        Eye color:
                        {character.eye_color}
                    </p>
                    <p id="height">
                        Height:
                        {character.height}
                    </p>
                    <p id="mass">
                        Mass:
                        {character.mass}
                    </p>
                    <p id="skin">
                        Skin color:
                        {character.skin_color}
                    </p>
                </Modal.Description>
            </Modal.Content>
        </Modal>
    );
};

CharacterModal.propTypes = {
    character: PropTypes.shape({
        name: PropTypes.string.isRequired,
        gender: PropTypes.string.isRequired,
        eye_color: PropTypes.string.isRequired,
        height: PropTypes.string.isRequired,
        mass: PropTypes.string.isRequired,
        skin_color: PropTypes.string.isRequired,
    }).isRequired,
    onCloseCallback: PropTypes.func.isRequired,
};


export default CharacterModal;
