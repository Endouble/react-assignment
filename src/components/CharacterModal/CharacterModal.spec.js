import React from 'react';
import { shallow, mount } from 'enzyme';

import CharacterModal from './CharacterModal';

describe('Character Modal Test', () => {
  let props;
  let shallowCharacterModal;

  const getShallowCharacterModal = () => {
    if (!shallowCharacterModal) {
      shallowCharacterModal = shallow(<CharacterModal {...props} />)
    }
    return shallowCharacterModal;
  };

  beforeEach(() => {
    props = {
      character: {
        name: "Luke Skywalker",
        gender: 'male',
        eye_color: 'blue',
        height: "172",
        mass: "77",
        skin_color: "fair"
      },
      onCloseCallback: jest.fn()
    };

    shallowCharacterModal = undefined;
  });

  test('Component should render without crashing', () => {
    expect(getShallowCharacterModal()).toBeDefined();
  });

  test('Component should render correctly', () => {
    expect(getShallowCharacterModal()).toMatchSnapshot();
  });

  test('Component should render character data', () => {
    const wrapper = mount(<CharacterModal {...props} />);
    const characterModalHeader = wrapper.find('.header');
    const characterModalContent = wrapper.find('.content .description');
    const expectedHeader = expect.stringMatching(/^.*Luke Skywalker/);
    const expectedContentGender = expect.stringMatching(/^.*Gender:male/);
    const expectedContentEyes = expect.stringMatching(/^.*Eye color:blue/);
    const expectedContentHeight = expect.stringMatching(/^.*Height:172/);
    const expectedContentMass = expect.stringMatching(/^.*Mass:77/);
    const expectedContentSkin = expect.stringMatching(/^.*Skin color:fair/);

    expect(characterModalHeader.text()).toEqual(expectedHeader);
    expect(characterModalContent.find('#gender').text()).toEqual(expectedContentGender);
    expect(characterModalContent.find('#eyes').text()).toEqual(expectedContentEyes);
    expect(characterModalContent.find('#height').text()).toEqual(expectedContentHeight);
    expect(characterModalContent.find('#mass').text()).toEqual(expectedContentMass);
    expect(characterModalContent.find('#skin').text()).toEqual(expectedContentSkin);
  });
})