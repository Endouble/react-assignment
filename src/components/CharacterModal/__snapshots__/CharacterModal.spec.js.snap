// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character Modal Test Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CharacterModal
    character={
        Object {
            "eye_color": "blue",
            "gender": "male",
            "height": "172",
            "mass": "77",
            "name": "Luke Skywalker",
            "skin_color": "fair",
          }
    }
    onCloseCallback={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "centered": true,
      "children": Array [
        <ModalHeader>
          Luke Skywalker
</ModalHeader>,
        <ModalContent>
          <ModalDescription>
                    <p
                              id="gender"
                    >
                              Gender:
                              male
                    </p>
                    <p
                              id="eyes"
                    >
                              Eye color:
                              blue
                    </p>
                    <p
                              id="height"
                    >
                              Height:
                              172
                    </p>
                    <p
                              id="mass"
                    >
                              Mass:
                              77
                    </p>
                    <p
                              id="skin"
                    >
                              Skin color:
                              fair
                    </p>
          </ModalDescription>
</ModalContent>,
      ],
      "closeIcon": true,
      "closeOnDimmerClick": true,
      "closeOnDocumentClick": false,
      "dimmer": true,
      "eventPool": "Modal",
      "onClose": [Function],
      "open": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Luke Skywalker",
        },
        "ref": null,
        "rendered": "Luke Skywalker",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ModalDescription>
            <p
                        id="gender"
            >
                        Gender:
                        male
            </p>
            <p
                        id="eyes"
            >
                        Eye color:
                        blue
            </p>
            <p
                        id="height"
            >
                        Height:
                        172
            </p>
            <p
                        id="mass"
            >
                        Mass:
                        77
            </p>
            <p
                        id="skin"
            >
                        Skin color:
                        fair
            </p>
</ModalDescription>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <p
                id="gender"
>
                Gender:
                male
</p>,
              <p
                id="eyes"
>
                Eye color:
                blue
</p>,
              <p
                id="height"
>
                Height:
                172
</p>,
              <p
                id="mass"
>
                Mass:
                77
</p>,
              <p
                id="skin"
>
                Skin color:
                fair
</p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Gender:",
                  "male",
                ],
                "id": "gender",
              },
              "ref": null,
              "rendered": Array [
                "Gender:",
                "male",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Eye color:",
                  "blue",
                ],
                "id": "eyes",
              },
              "ref": null,
              "rendered": Array [
                "Eye color:",
                "blue",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Height:",
                  "172",
                ],
                "id": "height",
              },
              "ref": null,
              "rendered": Array [
                "Height:",
                "172",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Mass:",
                  "77",
                ],
                "id": "mass",
              },
              "ref": null,
              "rendered": Array [
                "Mass:",
                "77",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Skin color:",
                  "fair",
                ],
                "id": "skin",
              },
              "ref": null,
              "rendered": Array [
                "Skin color:",
                "fair",
              ],
              "type": "p",
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "centered": true,
        "children": Array [
          <ModalHeader>
            Luke Skywalker
</ModalHeader>,
          <ModalContent>
            <ModalDescription>
                        <p
                                    id="gender"
                        >
                                    Gender:
                                    male
                        </p>
                        <p
                                    id="eyes"
                        >
                                    Eye color:
                                    blue
                        </p>
                        <p
                                    id="height"
                        >
                                    Height:
                                    172
                        </p>
                        <p
                                    id="mass"
                        >
                                    Mass:
                                    77
                        </p>
                        <p
                                    id="skin"
                        >
                                    Skin color:
                                    fair
                        </p>
            </ModalDescription>
</ModalContent>,
        ],
        "closeIcon": true,
        "closeOnDimmerClick": true,
        "closeOnDocumentClick": false,
        "dimmer": true,
        "eventPool": "Modal",
        "onClose": [Function],
        "open": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Luke Skywalker",
          },
          "ref": null,
          "rendered": "Luke Skywalker",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ModalDescription>
              <p
                            id="gender"
              >
                            Gender:
                            male
              </p>
              <p
                            id="eyes"
              >
                            Eye color:
                            blue
              </p>
              <p
                            id="height"
              >
                            Height:
                            172
              </p>
              <p
                            id="mass"
              >
                            Mass:
                            77
              </p>
              <p
                            id="skin"
              >
                            Skin color:
                            fair
              </p>
</ModalDescription>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <p
                  id="gender"
>
                  Gender:
                  male
</p>,
                <p
                  id="eyes"
>
                  Eye color:
                  blue
</p>,
                <p
                  id="height"
>
                  Height:
                  172
</p>,
                <p
                  id="mass"
>
                  Mass:
                  77
</p>,
                <p
                  id="skin"
>
                  Skin color:
                  fair
</p>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Gender:",
                    "male",
                  ],
                  "id": "gender",
                },
                "ref": null,
                "rendered": Array [
                  "Gender:",
                  "male",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Eye color:",
                    "blue",
                  ],
                  "id": "eyes",
                },
                "ref": null,
                "rendered": Array [
                  "Eye color:",
                  "blue",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Height:",
                    "172",
                  ],
                  "id": "height",
                },
                "ref": null,
                "rendered": Array [
                  "Height:",
                  "172",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Mass:",
                    "77",
                  ],
                  "id": "mass",
                },
                "ref": null,
                "rendered": Array [
                  "Mass:",
                  "77",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Skin color:",
                    "fair",
                  ],
                  "id": "skin",
                },
                "ref": null,
                "rendered": Array [
                  "Skin color:",
                  "fair",
                ],
                "type": "p",
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
